Shopify/cli
# Implement our own `path.cwd()` function to get the current working directory

### WHY are these changes introduced?

Fixes https://github.com/Shopify/cli/issues/1186

### WHAT is this pull request doing?

- Adds a new `path.cwd()` function that relies on `process.env.INIT_CWD`
- Adds a new eslint rule to advise against using `process.cwd()`

### How to test your changes?

- Add these lines to `packages/app/src/cli/commands/app/info.ts`
```
    // eslint-disable-next-line rulesdir/no-process-cwd
    output.info(`${process.cwd()} vs ${cwd()}`)
```
- Run `pnpm build`
- Create a new shopify app in this way `pnpm create-app --local --path ~/Desktop`
- Cd into the app folder and generate an extension `pnpm generate extension`
- From the extension folder run `pnpm app shopify info`
- At the top you should see the difference between `process.cwd()` and `path.cwd()`
- Remove the eslint disable comment to see the eslint rule work

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
